FlowCrypt/Controllers/Threads/ThreadDetailsViewController.swift

```diff
@@ -488,7 +488,12 @@ extension ThreadDetailsViewController: ASTableDelegate, ASTableDataSource {
                     msgAttachment: attachment,
                     index: attachmentIndex
                 ),
-                onDownloadTap: { [weak self] in self?.attachmentManager.open(attachment) }
+                onDownloadTap: { [weak self] in
+                    guard let self = self else { return }
+                    Task {
+                        await self.attachmentManager.open(attachment)
+                    }
```

Comment:
Also here errors should be handled with a modal.

Reply:
Error is handled in `open` method. Do you want to move it here? I guess this was done to avoid code duplication.
```
func open(_ attachment: MessageAttachment) async {
    do {
        try await filesManager.saveToFilesApp(file: attachment, from: self)
    } catch {
        controller?.showToast(
            "message_attachment_saved_with_error".localizeWithArguments(error.localizedDescription)
        )
    }
}
```